pkg/checkup/watch.go

```diff
@@ -0,0 +1,66 @@
+package checkup
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"time"
+
+	batchv1 "k8s.io/api/batch/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8swatch "k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
+)
+
+const JobNameLabel = "job-name"
+
+func WaitForJobToComplete(client *kubernetes.Clientset, jobName, jobNamespace string, timeout time.Duration) (*batchv1.Job, error) {
+	jobLabel := fmt.Sprintf("%s=%s", JobNameLabel, jobName)
+	jobWatcher, err := client.BatchV1().Jobs(jobNamespace).Watch(context.Background(), metav1.ListOptions{LabelSelector: jobLabel})
```

Comment:
AFAIK the usage of Watch() is not recommend.
IMO we should use an informer.

Reply:
Would you please share why Watch isnt recommended?

Note that executing a checkup is one shot operation, and there is only one instance of it.
Using an informer for the checkup Job is an overkill, requires some boilerplate and thus have higher maintenance.
For now I prefer to keep it simple for now and use Watch and later on consider switching to Informer.